libraries/AP_HAL/DSP.cpp

```diff
@@ -202,7 +202,7 @@ float DSP::calculate_quinns_second_estimator(const FFTWindowState* fft, const fl
     const float am = (complex_fft[k_m1] * complex_fft[k] + complex_fft[k_m1 + 1] * complex_fft[k + 1]) / divider;
 
     // sanity check
-    if (is_zero(ap) || is_zero(am)) {
+    if (is_equal(ap, 1.0f) || is_equal(am, 1.0f)) {
```

Comment:
I think it should check if (1.0 - ap) is below some minimum threshold, same for am

Reply:
Ok fixed. I'm not sure what an appropriate lower bound is, but I have some real values in a spreadsheet where am and ap are between 0.1 and 0.9 so 0.01 seems pretty safe.